### PR TITLE
ST_AsLatLonText: round minutes, #3688

### PR DESCRIPTION
**Pull request by mwtoews**
Fixes https://trac.osgeo.org/postgis/ticket/3688

### CODE REVIEW

**Comment by mwtoews** [comment]
Hmm, nope. This PR requires more work to ensure it doesn't bugger up the other unit tests.

**Comment by mwtoews** [comment]
Looks OK now, ready for review.

File: liblwgeom/cunit/cu_print.c [hunk 1]
```diff
@@ -80,6 +80,8 @@ static void test_lwprint_default_format(void)
 	test_lwprint_assert_format("POINT(-181 -91)",           ""  , "89\xC2\xB0""0'0.000\"S 1\xC2\xB0""0'0.000\"W");
 	test_lwprint_assert_format("POINT(-180.0001 -90.0001)", NULL, "89\xC2\xB0""59'59.640\"S 0\xC2\xB0""0'0.360\"W");
 	test_lwprint_assert_format("POINT(-2348982391.123456 -238749827.34879)", ""  , "12\xC2\xB0""39'4.356\"N 31\xC2\xB0""7'24.442\"W");
+    /* https://trac.osgeo.org/postgis/ticket/3688 */
```
Comment (strk) [inline liblwgeom/cunit/cu_print.c]: indent seems to be off here

**Comment by strk** [comment]
Since you're at it, squash-rebase and put your name in the commit log (authorship will be lost from GIT to SVN) - - Also mention the trac ticket (even better "close #ticketnum") -- Doing so would simplify the merge process.

Thanks!

**Comment by mwtoews** [comment]
@strk it should be all tidied up now

**Comment by strk** [comment]
Thanks - I had to add your name in the commit log (hint for next time) - eventual follow ups on trac: https://trac.osgeo.org/postgis/ticket/3688#comment:3

**Comment by robe2** [comment]
This needs more work.  Breaks my 32-bit windows runs.

**Comment by strk** [comment]
Can you drop a link to failure details ?

**Comment by robe2** [comment]
```
Suite: printing
  Test: test_lwprint_default_format ...
Assert failed:
	76Â°35'60.000"S 76Â°35'60.000"E	(actual)
	76Â°36'0.000"S 76Â°36'0.000"E	(expected)
FAILED
    1. cu_print.c:33  - CU_ASSERT_STRING_EQUAL(actual,expected)

```

I put on the ticket as well.  https://trac.osgeo.org/postgis/ticket/3688

Looks like it's returning the old answers.  64-bit windows seems fine though.

**Comment by robe2** [comment]
If this isn't fixed by the time I am ready to do 2.3 release, which I plan for next week, I'm going to have to

a) Back out this change from 2.3

or
b) disable this test for 2.3

It's probably safer to back it out since it could affect more than ST_AsLongLatText.

**Comment by mwtoews** [comment]
@robe2 I have an attempt to fix ready for you to test on a 32-bit Windows build from this PR. I've rebased this PR on svn-trunk and applied the new changes, so try it out from the generated patch/diff.

I don't have a Windows computer anymore, so I can't help test any further for this specific failure. Also, I wasn't able to see the failure on Linux with `-m32` to build 32-bit. I've implemented a direct method to round seconds and increment minutes. However, there is now a chance that minutes may be incremented past 59 to 60, unless I handle that case to round minutes and increment degrees.

I've removed the previous calculation `seconds = (val - (degrees + (minutes / 60))) * 3600;` since building with `-m32` formatted the seconds `-0.0`. The revised calculations use `modf`, which will not introduce negatives into the fractional part.

**Comment by robe2** [comment]
@mwtoews  Thanks,  I'll test in the next day or so.

**Comment by robe2** [comment]
Okay that seemed to do the trick.  I've committed to 2.3.  I'll close this after I commit to 2.4

**Pull request closed without merging**